### DISCUSSION TITLE
Time series "active" flag

### DISCUSSION THREAD

**Discussion by mdstanfill** [USACE/cwms-data-api#791] [General]
Is there a way to determine whether a time series is "active" in CDA?  With CWMS API, I would check the `net_ts_active_flag` of `cwms_v_ts_id2`, but I don't see a way to find this out with the existing endpoints (`catalog/TIMESERIES`, `timeseries/recent` or `timeseries`).

**Comment by MikeNeilson** [comment]
With the `/timeseries/identifier-descriptor` endpoint, though it doesn't allow searching.

But that field and the search ability should really be in both.

**Reply by mdstanfill** [reply]
Just what I needed!  Thanks!

**Reply by mdstanfill** [reply]
OK, that _looked_ like what I needed.  It _does_ return an active flag, I'm just not sure which one.  The one I'm interested in is `cwms_v_ts_id2.net_ts_active_flag`, as it's (to my limited understanding) a rollup of all the activity flags that can keep a time series from receiving data.

Sorry for the enormous delay on following up on this.  There's so much other stuff to do that I mentally marked this as resolved and postponed implementing it until I got more pressing matters out of the way.

Of note: if `catalog/TIMESERIES` supported filtering on active time series (again, the net or "rollup" active flag), this call would not be necessary in the application I'm trying to implement (and would not have to be called separately for each time series in the catalog).  That's probably the most efficient way of implementing a fix, though `timeseries/identifier-descriptor` might yet be improved by referencing `cwms_v_ts_id2.net_ts_active_flag` - but I don't know the use cases for active flags at other offices, and whether a specific active flag is desirable there.
